scripts/components/cart-item.jsx

```diff
@@ -0,0 +1,153 @@
+
+import { formatMoney } from '~mod/utils';
+
+export default class CartItem extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			editingVariant: false,
+		};
+
+		if (props.item.models && props.item.models.variantOptions) {
+			this.state.variantOptions = props.item.models.variantOptions;
+			this.state.selectedVariant = this.state.variantOptions.find((opt) => opt.id === props.item.id);
+		}
+	}
+
+	componentDidMount() {
+		const optionsEl = document.getElementById(`cart-item__variants__${this.props.item.id}`);
+		if (optionsEl) {
+			optionsEl.addEventListener('show.bs.collapse', this.toggleVariantEdit);
+			optionsEl.addEventListener('hidden.bs.collapse', this.toggleVariantEdit);
+		}
+	}
+
+	componentWillUnmount() {
+		const optionsEl = document.getElementById(`cart-item__variants__${this.props.item.id}`);
+		if (optionsEl) {
+			optionsEl.removeEventListener('show.bs.collapse', this.toggleVariantEdit);
+			optionsEl.removeEventListener('hidden.bs.collapse', this.toggleVariantEdit);
+		}
+	}
+
+	onSelectVariant(opt) {
+		if (opt.available && opt.id !== this.state.selectedVariant.id) {
+			this.setState({ selectedVariant: opt });
+		}
+	}
+
+	onRemoveItem = () => {
+		this.props.onRemoveItem(this.props.item);
+	}
+
+	toggleVariantEdit = () => {
+		if (this.state.editingVariant) {
+			this.setState({ editingVariant: false }, () => {
+				this.props.onChangeVariant(this.props.item, this.state.selectedVariant.id);
+			});
+		} else {
+			this.setState({ editingVariant: true });
+		}
+	}
+
+	render() {
+		const { item } = this.props;
+		const { models } = item;
+		return (
+			<li className="cart-item border-bottom">
+				<figure className="row py-2 mb-0 align-items-start">
+					<ConditionWrapper
+						condition={models.isFree}
+						wrapper={(children) => <a href={item.url} className="col-3">{children}</a>}
+						wrapperFalse={(children) => <div className="col-3">{children}</div>}
+					>
+						<picture>
+							<img src={models.image} className="w-100" alt={item.product_title} />
+						</picture>
+					</ConditionWrapper>
+					<figcaption className="col-9">
+						<div className="d-flex align-items-start">
+							<h4 className="flex-grow-1 mr-1">{item.product_title}</h4>
+							{!models.isFree && (<button className="cart-item__remove btn-unstyled sni sni__trash" type="button" aria-label="Remove" onClick={this.onRemoveItem} />)}
+						</div>
+
+						{models.color && models.variantOptions.length > 1 && (
+							<>
+								<div className="d-flex flex-wrap mb-1 align-items-center">
+									<i className={`swatch ${models.colorHandle}`} />
+									<i className="ml-1 mr-1">{models.color}</i>
+									{!models.isFree && (
+										<a className="cart-item__edit" data-toggle="collapse" href={`#cart-item__variants__${item.id}`} role="button" aria-expanded="false" aria-controls={`cart-item__variants__${item.id}`}>{this.state.editingVariant ? 'Done' : 'Edit'}</a>
```

Comment:
we should reduce the amount of HTML. I don't think we need to group element top and bottom, also we can reduce the tags to create the qty selector.